javalanglib/src/main/scala/java/lang/_String.scala

```diff
@@ -734,6 +734,215 @@ for (cp <- 0 to Character.MAX_CODE_POINT) {
+      case 1 if java.lang.Character.isWhitespace(charAt(0)) => ""
+      case n                                                =>
+        var idx = 0
+        while (idx < n && Character.isWhitespace(charAt(idx))) idx += 1
+        substring(idx)
+    }
+  }
+
+  def stripTrailing(): String = {
+    length() match {
+      case 0                                                => thisString
+      case 1 if java.lang.Character.isWhitespace(charAt(0)) => ""
+      case n                                                =>
+        var idx = n - 1
+        while (idx >= 0 && Character.isWhitespace(charAt(idx))) idx -= 1
+        substring(0, idx + 1)
+    }
+  }
+
+  def strip(): String = {
+    length() match {
+      case 0                                                => thisString
+      case 1 if java.lang.Character.isWhitespace(charAt(0)) => ""
+      case n                                                =>
+        var leading = 0
+        while (leading < n && Character.isWhitespace(charAt(leading))) leading += 1
+        if (leading == n) ""
+        else {
+          var trailing = n - 1
+          while (trailing >= 0 && Character.isWhitespace(charAt(trailing))) trailing -= 1
+          if (leading == 0 && trailing + 1 == n) thisString
+          else substring(leading, trailing + 1)
+        }
+    }
+  }
+
+  def isBlank(): scala.Boolean = {
+    val len = length()
+    var start = 0
+    while (start != len && java.lang.Character.isWhitespace(charAt(start))) start += 1
+    start == len
+  }
+
+  private def splitLines(): js.Array[String] = {
+    val xs   = js.Array[String]()
+    val len  = length()
+    var idx  = 0
+    var last = 0
+    while (idx < len) {
+      val c = charAt(idx)
+      if (c == '\n' || c == '\r') {
+        xs.push(substring(last, idx))
+        if (c == '\r' && idx + 1 < len && charAt(idx + 1) == '\n') idx += 1
+        last = idx + 1
+      }
+      idx += 1
+    }
+    // make sure we add the last segment, but not the last new line
+    if (last != len) xs.push(substring(last))
+    xs
+  }
+
+  def indent(n: Int): String = {
+
+    def forEachLn(f: js.Function1[String, String]): String = {
+      var out = ""
+      var i   = 0
+      val xs  = splitLines()
+      while (i < xs.length) {
+        out += f(xs(i)) + "\n"
+        i += 1
+      }
+      out
+    }
+
+    n match {
+      case remove if remove < 0 =>
+        forEachLn { l =>
+          var idx = 0
+          val lim = if (l.length > remove) l.length else remove
+          while (idx < lim && Character.isWhitespace(l.charAt(idx))) idx += 1
+          l.substring(idx)
+        }
+      case insert if insert > 0 =>
+        val padding = " ".asInstanceOf[_String].repeat(insert)
+        forEachLn(padding + _)
+      // as per spec, normalise new line for n == 0
+      case _ => forEachLn(l => l)
+    }
+  }
+
+  def stripIndent(): String = {
+    if (isEmpty()) {
+      ""
```

Comment:
It's needed because we attempt to identify the last new line which requires a non-empty string. I suppose we can fold this check in but it's not more readable.

Reply:
OK, let's keep it as is, then :)